pkg/resources/secret_with_basic_authentication.go

```diff
@@ -0,0 +1,163 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var secretBasicAuthenticationSchema = func() map[string]*schema.Schema {
+	secretBasicAuthentication := map[string]*schema.Schema{
+		"username": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the username value to store in the secret when setting the TYPE value to PASSWORD.",
+		},
+		"password": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the password value to store in the secret when setting the TYPE value to PASSWORD.",
```

Comment:
username and password should be probably sensitive (`Sensitive: true`), I think there was a documentation page where it's pointed what values should be treated as sensitive and what are not sensitive data. I think it was somewhat connected to user, @sfc-gh-asawicki you know what I'm talking about? 😅

Reply:
I agree with setting the password to be `Sensitive: true`, not really sure about the username since it is visible for both SHOW and DESCRIBE outputs as well

Reply:
I remember but I can't find it 🤷